packages/console/src/pages/SignInExperience/components/SignInMethodsForm.tsx

```diff
@@ -16,8 +16,40 @@ const signInMethods = Object.values(SignInMethodKey);
 
 const SignInMethodsForm = () => {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
-  const { register, watch, control } = useFormContext<SignInExperienceForm>();
+  const { register, watch, control, setValue } = useFormContext<SignInExperienceForm>();
   const primaryMethod = watch('signInMethods.primary');
+  const enableSecondary = watch('signInMethods.enableSecondary');
+  const social = watch('signInMethods.social');
+
+  useEffect(() => {
+    if (primaryMethod) {
+      // Set secondary method to false.
+      setValue(`signInMethods.${primaryMethod}`, false);
```

Comment:
Does this comment match the code? Looks like you are setting `primaryMethod` to false

Reply:
better?
```
// If one of the sign-in methods is primary, it should be unable to be set as secondary.
```

Reply:
non-identical guard

Reply:
updated